ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/application/plugins/api/PluginBuilder.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+@file:Suppress("DEPRECATION")
+
+package io.ktor.server.application.plugins.api
+
+import io.ktor.server.application.*
+import io.ktor.server.application.plugins.api.debug.*
+import io.ktor.server.config.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import io.ktor.util.*
+import io.ktor.util.debug.*
+import io.ktor.util.pipeline.*
+import kotlinx.coroutines.*
+import kotlin.random.*
+
+/**
+ * A builder that is available inside a plugin creation block. It allows you to define handlers for different stages
+ * (a.k.a. phases) of the HTTP pipeline.
+ **/
+@PluginsDslMarker
+public interface PluginBuilder<PluginConfig : Any> {
+
+    public val application: Application
+
+    /**
+     * Specifies how to modify HTTP call handling for the current [ApplicationPluginBuilder].
+     * @see OnCall
+     **/
+    public fun handle(block: suspend HandleContext<PluginConfig>.(call: ApplicationCall) -> Unit)
```

Comment:
Sorry, I don't like this name.
Let's consider the usage:
```
val p  = createPlugin(...) {
    handle { ... } // Handle what? Handle call? or handle something else? Why is is different from `onCall` then? 
    onCall { ... }
    onCallRespond { ... }
}
```
I prefer more explicit name and let's also discuss with Hadi?

Reply:
PS:
Will `onCall` be executed even if the call was not handled?
If so -- maybe let's rename them to `onCallHandled` and `onCallReceived` ?